### PR TITLE
Added support for scheduled maintenance check

### PR DESCRIPTION
**Pull request by ajmine**

### CODE REVIEW

File: Source/GTMaintenanceHelper.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        isUndergoingMaintenance = (data?["under_maintenance"] as? [String: AnyObject])?["ios"] as? Bool ?? false
+    }
+
+    //
+    // MARK : Polling
+    //
+
+    private func beginPolling() {
+        guard #available(iOS 10.0, *) else { return }
+        pollingTimer = Timer.scheduledTimer(withTimeInterval: 30, repeats: true, block: poll())
```
Comment (mattbanach) [inline Source/GTMaintenanceHelper.swift]: shame you can't use GAMDelayHelper

Comment (ajmine) [reply Source/GTMaintenanceHelper.swift]: Hah yea. I should make the time interval into a variable

File: Source/GTMaintenanceHelper.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+            self?.isUndergoingMaintenance = (responseObject?["under_maintenance"] as? [String: AnyObject])?["ios"] as? Bool ?? false
+            completion()
+        })
+        
+        dataTask.resume()
+    }
+
+    internal func updateWithData(_ data: [String: AnyObject]?) {
+        isUndergoingMaintenance = (data?["under_maintenance"] as? [String: AnyObject])?["ios"] as? Bool ?? false
```
Comment (MikeSilvis) [inline Source/GTMaintenanceHelper.swift]: it'd probably be nice to be able to pass down more useful information about what is being updated. Although we could simply use the string interceptions to do this

**Comment by ajmine** [comment]
p.s. the GTConstants PR needs to be merged first and released

**Review by MikeSilvis — APPROVED** [review]

**Pull request merged**